cmd/prow-job-dispatcher/main.go

```diff
@@ -329,20 +352,51 @@ func dispatchJobs(ctx context.Context, prowJobConfigDir string, maxConcurrency i
 	return utilerrors.NewAggregate(errs)
 }
 
+func runAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string, gitOnly bool) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
+
+	if !gitOnly {
+		logrus.Infof("Running: %s", fullCommand)
+		if err := bumper.Call(stdout, stderr, cmd, args...); err != nil {
+			return false, fmt.Errorf("failed to run %s: %w", fullCommand, err)
+		}
+	}
+
+	changed, err := bumper.HasChanges()
+	if err != nil {
+		return false, fmt.Errorf("error occurred when checking changes: %w", err)
+	}
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
+	}
+
+	gitCmd := "git"
+	if err := bumper.Call(stdout, stderr, gitCmd, []string{"add", "."}...); err != nil {
+		return false, fmt.Errorf("failed to 'git add .': %w", err)
+	}
+
+	commitArgs := []string{"commit", "-m", fullCommand, "--author", author}
+	if err := bumper.Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
+		return false, fmt.Errorf("fail to %s %s: %w", gitCmd, strings.Join(commitArgs, " "), err)
+	}
+
+	return true, nil
+}
+
 func main() {
-	secrets := sets.NewString()
-	logrus.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, func() sets.String {
-		return secrets
-	}))
 	o := gatherOptions()
 	if err := o.validate(); err != nil {
 		logrus.WithError(err).Fatal("Failed to complete options.")
 	}
-	if err := o.complete(&secrets); err != nil {
-		logrus.WithError(err).Fatal("Failed to complete options.")
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.PRCreationOptions.GitHubOptions.TokenPath, o.PrometheusOptions.PrometheusPasswordPath}); err != nil {
```

Comment:
I don't think the `TokenPath` should always be mandatory, only `if o.CratePR`

Reply:
Updated.